File: rb/features.rb
# TemporaryEmailService SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module TemporaryEmailServiceFeatures
  def self.make_feature(name)
    case name
    when "base"
      TemporaryEmailServiceBaseFeature.new
    when "test"
      TemporaryEmailServiceTestFeature.new
    else
      TemporaryEmailServiceBaseFeature.new
    end
  end
end

File: py/features.py
# TemporaryEmailService SDK feature factory

from feature.base_feature import TemporaryEmailServiceBaseFeature
from feature.test_feature import TemporaryEmailServiceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TemporaryEmailServiceBaseFeature(),
        "test": lambda: TemporaryEmailServiceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
